src/PrestaShopBundle/Controller/Admin/ModuleController.php

```diff
@@ -403,16 +405,24 @@ public function importModuleAction(Request $request)
     protected function installModule($module_name)
     {
         $modulesProvider = $this->container->get('prestashop.core.admin.data_provider.module_interface');
+
         if (! $modulesProvider->isModuleOnDisk($module_name)) {
             $modulesProvider->setModuleOnDiskFromAddons($module_name);
         }
 
-        $module = $modulesProvider->getModule($module_name);
-        $status = $module->install();
+        try {
+            $module = $modulesProvider->getModule($module_name);
+            $status = $module->install();
+        } catch (Exception $e) {
+            $status = false;
+            $module = $modulesProvider->getModule($module_name);
+        }
+
         if ($status) {
             $msg = sprintf('Module %s is now installed', $module_name);
         } else {
             $msg = sprintf('Could not install module %s (Additionnal Information: %s)', $module_name, join(', ', $module->getErrors()));
```

Comment:
`$module` has been defined in the try block, are you sure it will still exist here ?